Adaptive Pathfinding in Partially Observable Dynamic Environments

Avanatiker opened the pull request:
Introducing an algorithm solving adaptive pathfinding in partially observable dynamic environments using spatial partitioning, iterative pathing using Lazy D* Lite and clearance based path refining using Lazy θ* and Catmull Rom splines.

ToDo:
- Any angle not yet optimal. Look ahead may go too far and lengthen another shortcut next

TheBheta commented:
breathe

ToxicAven commented:
> breathe

I did, took me a minute

czho commented:
We might want to make the actions repo be in the github organization.

ToxicAven commented:
> We might want to make the actions repo be in the github organization.

No, as this is a personal actions run I made outside of the lambda project.

Avanatiker commented:
Thanks a lot :) cool tool

TheBheta commented:
your hard work paid off

Avanatiker committed:
Added pathfinder

Avanatiker committed:
Kinda working pathing

Avanatiker committed:
NodeTypes and different moves

Avanatiker committed:
Theta*

Avanatiker committed:
Better support check and config cleanup

Avanatiker committed:
Some fixes

Avanatiker committed:
Change target pos

Avanatiker committed:
Test lib

Avanatiker committed:
D* Lite - 2D with tests

Avanatiker committed:
3D Implementation

Avanatiker committed:
Merge branch 'feature/dstar-lite' into feature/pathfinder

Avanatiker committed:
Speed up clearance checks

Avanatiker committed:
More settings

Avanatiker committed:
Lazy D* Lite

Avanatiker committed:
Incremental pathing on state updates and more settings

Avanatiker committed:
Funny path renderer

Avanatiker committed:
D* cut off timeout

Avanatiker committed:
Merge branch 'master' into feature/pathfinder

Avanatiker committed:
Optimized threading and graph renderer

Avanatiker committed:
Pathing command, detailed graph renderer, graph invalidation

Avanatiker committed:
Fix refining of path start

Avanatiker committed:
Less frequent pathing updates

Avanatiker committed:
Add render setting functionalities

Avanatiker committed:
Improved renderer

Avanatiker committed:
Vertex invalidation approach

Avanatiker committed:
Incremental pathfinding with updateable weights

Avanatiker committed:
Node invalidation on world state update

Avanatiker committed:
Simplify invalidation

Avanatiker committed:
Test graph connectivity on node invalidation

Avanatiker committed:
Update rhs of newly created nodes

Avanatiker committed:
Clean move finder cache

Avanatiker committed:
Graph pruning on invalidation, graph consistency checks

Avanatiker committed:
Proper consistency test

Avanatiker committed:
Add path consistency tests and graph pruning improvements

Introduce extensive path consistency tests for scenarios with different graph connectivities and blocked nodes using D* Lite. Add graph pruning logic to exclude nodes marked as blocked and enhance debugging utilities for path validation.

Avanatiker committed:
Refactor graph connectivity logic and improve block update handling.

Abstracted grid connectivity into reusable methods for cleaner and more flexible code. Enhanced block update listener to skip redundant updates and improved logging clarity. Simplified pathfinding and traversal checks for better maintainability.

Avanatiker committed:
Merge remote-tracking branch 'origin/master' into feature/pathfinder

Avanatiker committed:
Improve test quality

Avanatiker committed:
Slight code improvements

Avanatiker committed:
Incremental package

Avanatiker committed:
Bidirectional graph initialization

Avanatiker committed:
Fix concurrency on graph render, relax path condition

Avanatiker committed:
Dynamic connectivity

Avanatiker committed:
Merge remote-tracking branch 'origin/master' into feature/pathfinder
